src/wobbly-effect.cpp

```diff
@@ -0,0 +1,584 @@
+                                                        actor,
+                                                        &actor_width,
+                                                        &actor_height);
+
+        priv->model = new wobbly::Model (wobbly::Point (0, 0),
+                                         actor_width,
+                                         actor_height,
+                                         priv->model_settings);
+
+        priv->width_changed_signal =
+            g_signal_connect_object (actor,
+                                     "notify::width",
+                                     G_CALLBACK (endless_shell_fx_wobbly_size_changed),
+                                     wobbly_effect,
+                                     static_cast <GConnectFlags> (G_CONNECT_AFTER));
+        priv->height_changed_signal =
+            g_signal_connect_object (actor,
+                                     "notify::height",
+                                     G_CALLBACK (endless_shell_fx_wobbly_size_changed),
+                                     wobbly_effect,
+                                     static_cast <GConnectFlags> (G_CONNECT_AFTER));
+    }
+
+    /* Whatever the actor, ensure that the effect is disabled at this point */
+    clutter_actor_meta_set_enabled (actor_meta,
+                                    FALSE);
+}
+
+static void
+endless_shell_fx_wobbly_set_property (GObject      *object,
+                                      guint        prop_id,
+                                      const GValue *value,
+                                      GParamSpec   *pspec)
+{
+    EndlessShellFXWobbly *wobbly_effect =
+        reinterpret_cast <EndlessShellFXWobbly *> (object);
+    EndlessShellFXWobblyPrivate *priv =
+        reinterpret_cast <EndlessShellFXWobblyPrivate *> (endless_shell_fx_wobbly_get_instance_private (wobbly_effect));
+
+    switch (prop_id)
+    {
+        case PROP_SPRING_K:
+            priv->model_settings.springConstant = g_value_get_double (value);
+            break;
+        case PROP_FRICTION:
+            priv->model_settings.friction = g_value_get_double (value);
+            break;
+        case PROP_SLOWDOWN_FACTOR:
+            priv->slowdown_factor = g_value_get_double (value);
+            break;
+        case PROP_OBJECT_MOVEMENT_RANGE:
+            priv->model_settings.maximumRange = g_value_get_double (value);
+            break;
+        default:
+            G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+            break;
+    }
+}
+
+static void
+endless_shell_fx_wobbly_finalize (GObject *object)
+{
+    EndlessShellFXWobbly *wobbly_effect =
+        reinterpret_cast <EndlessShellFXWobbly *> (object);
+    EndlessShellFXWobblyPrivate *priv =
+        reinterpret_cast <EndlessShellFXWobblyPrivate *> (endless_shell_fx_wobbly_get_instance_private (wobbly_effect));
+
+    if (priv->model)
+        delete priv->model;
+
+    if (priv->timeout_id)
+    {
```

Comment:
I'd love it if we could use an uniform coding style with the rest of `src` for this file.